src/App.tsx

```diff
@@ -1,6 +1,31 @@
+import { useEffect, useRef } from "react";
 import { BirthdayCountdown } from "./components/Countdown";
+import party from 'party-js';
 
 function App() {
+    const isConfettiOn = useRef(false);
+
+    useEffect(() => {
+      const shootConfetti = (event: MouseEvent) => {
+        if (!isConfettiOn.current) {
+          isConfettiOn.current = true;
+  
+          party.confetti(event, {
+            count: party.variation.range(0, 5),
+            spread: 100,
+          });
+  
+          requestAnimationFrame(() => {
+            isConfettiOn.current = false;
+          });
+        }
+      };
+  
+      document.addEventListener('mousemove', shootConfetti);
+      return () => {
+        document.removeEventListener('mousemove', shootConfetti);
+      };
+    }, []);
```

Comment:
I recommend moving this to its own file.

Here is a reference for how you can do so from react docs.

```
// useConfetti.ts
export function useConfetti() {
  const isConfettiOn = useRef(false);
  useEffect(() => {
    // all your logic
  }, []);
}
```

Reply:
oh very interesting. i will indeed do this practice. thank you for your review

Reply:
resolved